src/Microsoft.Health.Checkpoints/ICheckpoint.cs

```diff
@@ -0,0 +1,18 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Health.Checkpoints
+{
+    public interface ICheckpoint
+    {
+        string Identifier { get; set; }
+
+        DateTimeOffset LastProcessedDt { get; set; }
+
+        string LastProcessedId { get; set; }
+    }
```

Comment:
Suggestion: Thoughts on adding a partition concept here and using it in the path to the underlying blob?  Right now we will only have 1 but in the future we will likely have multiple agents each with their own affinity.

Reply:
This could be handled in how the identifier is generated but it seems cleaner to include it natively.

Reply:
I had partition originally but realized it was kind of redundant since the the identifier can include the partition. But will add partition back so that the two concepts of partition and id are obvious to the user.

Reply:
I am still on the fence about including a separate property for partition vs. just have an identifier with includes the partition + id. 

I would prefer to have one argument in ICheckpointClient here:
Task<ICheckpoint> GetCheckpointAsync(string checkpointIdentifier);

If we add partition, then I think we'd have to add an additional argument to the ICheckpointClient interface:
Task<ICheckpoint> GetCheckpointAsync(string checkpointIdentifier, string partitionId);

But then we'd just end up concat-ing them later when we do the blob lookup. Is updating the interface to have two argument preferred?

Reply:
I guess the real question is how will this be used up stream.  We could remove partition as a concept in the ICheckpoint itself.  Instead we could have another abstraction on top of the ICheckpointClient, i.e. PartitionedStorageCheckpointClient and potentially an additional interface, IParitionedCheckpointClient that introduces a required PartitionId property.  This class would have the responsibility of appending the partition id to the base identifier.  Thoughts?  If that approach sounds good we don't need to add it yet but is probably something we will add once we build on top of this for the IoT Connector version.

My main concern is I don't want partition management to be arbitrary string concats from the code calling this client.  When we do handle partitions I would like it to be well defined.

Reply:
I went ahead and added partition as part of the interface here so that we don't need to do another PartitionedStorageCheckpointClient in the future.